apps/voucher/hooks/useRealTimePrice.ts

```diff
@@ -0,0 +1,217 @@
+import React, { useState, useEffect } from "react"
```

Comment:
maybe with this PR merged we don't need this file anymore

https://github.com/GaloyMoney/galoy/commit/14b56dc3bf8ae84628568b406e2d11c1e2ab65c4

Reply:
Maybe we need ? We are using subscription API here as well.

Reply:
if we are using the subcription API to get the price, I think we don't need that anymore with the new query above

Reply:
Not sure if it's ready to use.
```
curl 'http://localhost:4455/graphql' \
  -H 'Accept-Language: en-US,en;q=0.9,ko;q=0.8' \
  -H 'Connection: keep-alive' \
  -H 'DNT: 1' \
  -H 'Origin: http://localhost:3006' \
  -H 'Referer: http://localhost:3006/' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-site' \
  -H 'User-Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 16_6 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.6 Mobile/15E148 Safari/604.1' \
  -H 'accept: */*' \
  -H 'content-type: application/json' \
  --data-raw $'{"operationName":"CurrencyConversionEstimation","variables":{"amount":68.85,"currency":"USD"},"query":"query CurrencyConversionEstimation($amount: Float\u0021, $currency: DisplayCurrency\u0021) {\\n  currencyConversionEstimation(amount: $amount, currency: $currency) {\\n    btcSatAmount\\n    id\\n    timestamp\\n    usdCentAmount\\n    __typename\\n  }\\n}"}'
{"data":null,"errors":[{"message":"Unexpected error occurred, please try again or contact support if it persists (code: BigIntFloatConversionError: 6884.999999999999)","locations":[{"line":1,"column":89}],"path":["currencyConversionEstimation"]}]}%                                                            
```

Also, is it the best option for the frontend? Every amount change calls an API, which takes a few milliseconds to update the amount. It does not look as smooth on the frontend, maybe better for backend ?

Reply:
@dolcalmi is there an issue with `CurrencyConversionEstimation`?

Reply:
@siddhart1o1 it's made for frontends. the idea is that you want to have the amount and then you do the conversation.

Reply:
<img width="461" alt="image" src="https://github.com/GaloyMoney/galoy/assets/59279771/81c19429-2ba7-45e4-b4ea-a98d598cad0a">

Still not sure if this is the right query to use here, as we are displaying the voucher amount `(voucher_amount = input amount - commission)` in USD. For this, we convert the user's currency to USD, for example, EUR is converted to USD. 
The voucher amount is updated with every change in the input amount (numpad click), so we need to calculate the voucher amount on every click, i.e., convert the user amount to USD.
And as this voucher amount is updated with every Numpad click. If we use this new query here, every change/click in numpad will cause an API call. Whereas, the `useRealTimePrice` hook provides base and offset values which we are using to calculate the price on the client and these offeset values are updated by the subscription API when needed.

Anyway, I've updated this to use `CurrencyConversionEstimation` maybe I'm over-optimizing it or missing something.